jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryEndpointCaller.java

```diff
@@ -161,14 +205,16 @@ private T call(RequestState requestState) throws IOException, RegistryException
         }
       }
 
+    } catch (HttpHostConnectException | SSLPeerUnverifiedException ex) {
```

Comment:
I'm not sure about this, but is it possible users might not want this insecure fallback mechanism?  What if they always want to use https or fail?

Reply:
Hmm, yea it seems that we might want to have the user enable the fallback to HTTP, since Docker only seems to allow insecure registries with a flag.

https://docs.docker.com/registry/insecure/